addon/components/high-charts.js

```diff
@@ -39,40 +39,24 @@ export default Component.extend({
   didReceiveAttrs() {
     this._super(...arguments);
 
-    if (!(get(this, 'content') && get(this, 'chart'))) {
-      return;
-    }
+    const content = get(this, 'content');
+    const chart = get(this, 'chart');
+    if (!content || !chart) return;
 
-    let chart = get(this, 'chart');
     let noData = chart.get('noData');
+    if (noData != null) noData.remove();
 
-    if (noData != null) {
-      noData.remove();
-    }
-
-    let numToRemove = chart.series.length - get(this, 'content').length;
-
-    for (let i = numToRemove; i > 0; i--) {
-
-      let lastIndex = chart.series.length - 1;
-
-      if (chart.series[lastIndex]) {
-        chart.series[lastIndex].remove(false);
-      }
-
-    }
-
-    get(this, 'content').forEach((series, idx) => {
-
-      if (chart.series[idx]) {
-        return chart.series[idx].setData(series.data, false);
-      } else {
-        return chart.addSeries(series, false);
-      }
+    // remove series that aren't navigator
+    const seriesToRemove = chart.series.filter((series) => {
+      if (series.name === 'Navigator') return false;
+      return true;
     });
+    seriesToRemove.forEach((series) => series.remove(false));
 
-    return chart.redraw();
+    // readd and add new series
+    content.forEach((series) => chart.addSeries(series, false));
 
```

Comment:
I think we need to add an extra step here to refresh the navigator for a highstock chart. This code seems to refresh the navigator:
```js
chart.xAxis[0].setExtremes();
```

This post explains in more detail: http://stackoverflow.com/questions/18566314/highstock-navigator-time-range-does-not-update-after-adding-removing-series

Adding this line updates the navigator which you can see happen in the tests dummy app when clicking the "Update Series Data" button.